### PR TITLE
Spring WebMVC

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,8 +5,9 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>org.example</groupId>
-    <artifactId>My-own-project2-Cinema</artifactId>
+    <artifactId>String.inrto</artifactId>
```
Comment (jah701) [inline pom.xml]: spring? :)

Comment (jah701) [reply pom.xml]: I get the same issue with "sprint" in my case :D

File: pom.xml [hunk 2]
```diff
@@ -49,12 +56,35 @@
             <artifactId>mysql-connector-java</artifactId>
             <version>8.0.21</version>
         </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
```
Comment (jah701) [inline pom.xml:60]: I'm not sure if we have to keep this dependencies for this h\w (jstl, jsp, servlet) 🤔

File: src/main/java/spring/intro/controller/HelloController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package spring.intro.controller;
```
Comment (jah701) [inline src/main/java/spring/intro/controller/HelloController.java]: I suppose we've had to delete this class

File: src/main/java/spring/intro/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @GetMapping("/{id}")
+    public UserResponseDto getUserById(@PathVariable Long id) {
+        logger.info("Trying to find user by id: " + id + " from DB in UserController");
+        User user = userService.getById(id);
+        return convertUserToUserDto(user);
+    }
+
+    @GetMapping("/users")
```
Comment (jah701) [inline src/main/java/spring/intro/controller/UserController.java]: I guess we have to keep it without value acoording to checklist
```method List<UserResponseDto> getAll, URL: GET: /user/```

**Review by jah701 — APPROVED** [review]
Well done!

File: src/main/java/spring/intro/config/WebConfig.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {"spring.intro.controller"})
+public class WebConfig {
+
+    @Bean
+    public InternalResourceViewResolver resolver() {
```
Comment (Sofasmile) [inline src/main/java/spring/intro/config/WebConfig.java]: let's remove this bean

File: src/main/java/spring/intro/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import spring.intro.config.AppConfig;
+import spring.intro.dto.UserResponseDto;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private static final Logger logger = Logger.getLogger(UserController.class);
+    AnnotationConfigApplicationContext context
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controller/UserController.java]: let's use constructor injection here

File: pom.xml (hunk 1 above)
Comment (KostiantynVorobiov) [reply pom.xml]: :)))Done

File: src/main/webapp/WEB-INF/views/hello.jsp [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/hello.jsp]: let's remove jsp page

File: pom.xml (hunk 2 above)
Comment (KostiantynVorobiov) [reply pom.xml:60]: Done

File: src/main/java/spring/intro/controller/HelloController.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/spring/intro/controller/HelloController.java]: Done

File: src/main/java/spring/intro/controller/UserController.java (hunk 4 above)
Comment (KostiantynVorobiov) [reply src/main/java/spring/intro/controller/UserController.java]: Done

File: src/main/java/spring/intro/config/WebConfig.java (hunk 5 above)
Comment (KostiantynVorobiov) [reply src/main/java/spring/intro/config/WebConfig.java]: Done

File: src/main/webapp/WEB-INF/views/hello.jsp (hunk 7 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/WEB-INF/views/hello.jsp]: Done

File: src/main/java/spring/intro/controller/UserController.java [hunk 8]
```diff
@@ -17,9 +16,12 @@
 @RequestMapping("/user")
 public class UserController {
     private static final Logger logger = Logger.getLogger(UserController.class);
-    AnnotationConfigApplicationContext context
-            = new AnnotationConfigApplicationContext(AppConfig.class);
-    UserService userService = context.getBean(UserService.class);
+    private UserService userService;
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controller/UserController.java]:
```suggestion
    private final UserService userService;
```

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/spring/intro/controller/UserController.java (hunk 8 above)
Comment (KostiantynVorobiov) [reply src/main/java/spring/intro/controller/UserController.java]: Done

**Pull request merged**